### PR TITLE
mpc: 0.34 -> 0.35

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31125,9 +31125,7 @@ with pkgs;
     withConplay = false;
   };
 
-  mpc-cli = callPackage ../applications/audio/mpc {
-    inherit (python3Packages) sphinx;
-  };
+  mpc-cli = mpc;
```
Comment (Scrumplex) [inline pkgs/top-level/all-packages.nix:31128]: I think this should just go into `aliases.nix`?

Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:31128]: Hum, OK.

**Review by Scrumplex — CHANGES_REQUESTED** [review]
Some derivations still reference mpc-cli. Those should be updated so that Nixpkgs can evaluate with aliases disabled.

```console
$ rg --files-with-matches mpc-cli
nixos/tests/mpd.nix
nixos/modules/services/hardware/triggerhappy.nix
pkgs/top-level/aliases.nix
pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix
pkgs/applications/audio/clerk/default.nix
```

**Comment by AndersonTorres** [comment]
I will revert the aliasing.

**Comment by Scrumplex** [comment]
Git diff to use `aliases.nix` and fix the remaining dependents:

```diff
diff --git a/nixos/modules/services/hardware/triggerhappy.nix b/nixos/modules/services/hardware/triggerhappy.nix
index d2137971b3a7..26dec0a29c66 100644
--- a/nixos/modules/services/hardware/triggerhappy.nix
+++ b/nixos/modules/services/hardware/triggerhappy.nix
@@ -67,7 +67,7 @@ in
         type = lib.types.listOf (lib.types.submodule bindingCfg);
         default = [];
         example = lib.literalExpression ''
-          [ { keys = ["PLAYPAUSE"];  cmd = "''${pkgs.mpc-cli}/bin/mpc -q toggle"; } ]
+          [ { keys = ["PLAYPAUSE"];  cmd = "''${pkgs.mpc}/bin/mpc -q toggle"; } ]
         '';
         description = ''
           Key bindings for {command}`triggerhappy`.
diff --git a/nixos/tests/mpd.nix b/nixos/tests/mpd.nix
index 0772c05d12ac..8e3ea867bfaa 100644
--- a/nixos/tests/mpd.nix
+++ b/nixos/tests/mpd.nix
@@ -95,7 +95,7 @@ import ./make-test-python.nix ({ pkgs, lib, ... }:
     };
 
   testScript = ''
-    mpc = "${pkgs.mpc-cli}/bin/mpc --wait"
+    mpc = "${pkgs.mpc}/bin/mpc --wait"
 
     # Connects to the given server and attempts to play a tune.
     def play_some_music(server):
diff --git a/pkgs/applications/audio/clerk/default.nix b/pkgs/applications/audio/clerk/default.nix
index f77b03daac78..4eebd96f516a 100644
--- a/pkgs/applications/audio/clerk/default.nix
+++ b/pkgs/applications/audio/clerk/default.nix
@@ -5,7 +5,7 @@
 , rofi
 , tmux
 , fzf
-, mpc-cli
+, mpc
 , perl
 , util-linux
 , libnotify
@@ -59,7 +59,7 @@ stdenv.mkDerivation {
   postFixup = let
     binPath = lib.makeBinPath [
       libnotify
-      mpc-cli
+      mpc
       rofi
       tmux
       fzf
diff --git a/pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix b/pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix
index 5810070d2ff4..d5567ce15e8f 100644
--- a/pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix
+++ b/pkgs/applications/window-managers/i3/bumblebee-status/plugins.nix
@@ -89,7 +89,7 @@ in
   memory.propagatedBuildInputs = [ pkgs.gnome-system-monitor ];
   messagereceiver = { };
   mocp.propagatedBuildInputs = [ pkgs.moc ];
-  mpd.propagatedBuildInputs = [ pkgs.mpc-cli ];
+  mpd.propagatedBuildInputs = [ pkgs.mpc ];
   network.propagatedBuildInputs = [ py.netifaces pkgs.iw ];
   network_traffic.propagatedBuildInputs = [ py.netifaces ];
   nic.propagatedBuildInputs = [ py.netifaces pkgs.iw ];
diff --git a/pkgs/top-level/aliases.nix b/pkgs/top-level/aliases.nix
index 087dab080f6b..31d1561f294f 100644
--- a/pkgs/top-level/aliases.nix
+++ b/pkgs/top-level/aliases.nix
@@ -1072,7 +1072,8 @@ mapAliases {
   mozart-binary = throw "'mozart-binary' has been renamed to/replaced by 'mozart2-binary'"; # Converted to throw 2023-09-10
   mozart = throw "'mozart' has been renamed to/replaced by 'mozart2-binary'"; # Converted to throw 2023-09-10
   mp3info = throw "'mp3info' has been removed due to lack of maintenance upstream. Consider using 'eartag' or 'tagger' instead"; # Added 2024-09-14
-  mpc_cli = mpc-cli; # moved from top-level 2022-01-24
+  mpc-cli = mpc; # Added 2024-10-14
+  mpc_cli = mpc; # moved from top-level 2022-01-24
   mpd_clientlib = libmpdclient; # Added 2021-02-11
   mpdevil = plattenalbum; # Added 2024-05-22
   mpg321 = throw "'mpg321' has been removed due to it being unmaintained by upstream. Consider using mpg123 instead."; # Added 2024-05-10
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 9deb23803ee4..16a7d86ae54a 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -31077,8 +31077,6 @@ with pkgs;
     withConplay = false;
   };
 
-  mpc-cli = mpc; # green-alias added 2024-10-14
-
   clerk = callPackage ../applications/audio/clerk { };
 
   nbstripout = callPackage ../applications/version-management/nbstripout { };

```

**Comment by AndersonTorres** [comment]
OK, let's edit this.

Converting to draft.

**Review by Scrumplex — COMMENTED** [review]
Changes LGTM

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347387`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>clerk</li>
    <li>mpc</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Comment by Scrumplex** [comment]
IMO "green aliases" make sense when you have different established spellings (i.e. `network-manager` and `NetworkManager`) or versioned dependencies (i.e. `glfw` and `glfw3`).

I personally never understood why it was called `mpc-cli` before, as every other distro I know of calls it `mpc`. I think we should discourage the use of other package names, as `mpc` seems to be the established standard.

**Comment by AndersonTorres** [comment]
I think it was a historic accident. `mpd_clientlib` was a similar case.

**Comment by Scrumplex** [comment]
I think the throw should be added after the 24.11 branch off.

**Comment by AndersonTorres** [comment]
The previous throw, that uses underscores, is too old. 
It would make few sense to keep one and not the other.

File: pkgs/by-name/cl/clerk/package.nix [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  rofi,
+  stdenv,
+  tmux,
+  unstableGitUpdater,
+  util-linux,
+}:
+
+stdenv.mkDerivation {
+  pname = "clerk";
+  version = "0-unstable-2023-10-07";
```
Comment (wineee) [inline pkgs/by-name/cl/clerk/package.nix:19]:
```suggestion
  version = "4.0.5-unstable-2023-10-07";
```

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/top-level/aliases.nix [hunk 3]
```diff
@@ -755,7 +755,8 @@ mapAliases {
   mongodb-5_0 = throw "mongodb-5_0 has been removed, it's end of life since October 2024"; # Added 2024-10-01
   moz-phab = mozphab; # Added 2022-08-09
   mp3info = throw "'mp3info' has been removed due to lack of maintenance upstream. Consider using 'eartag' or 'tagger' instead"; # Added 2024-09-14
-  mpc_cli = mpc-cli; # moved from top-level 2022-01-24
+  mpc-cli = throw "mpc-cli has been renamed to mpc."; # Added 2024-10-14
+  mpc_cli = throw "mpc_cli has been renamed to mpc."; # Added 2024-10-14
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/top-level/aliases.nix]: Done

File: pkgs/by-name/cl/clerk/package.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/by-name/cl/clerk/package.nix:19]: Done

**Review by wineee — APPROVED** [review]

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347387`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>clerk</li>
    <li>mpc</li>
  </ul>
</details>

**Review by Scrumplex — COMMENTED** [review]
Changes LGTM

File: nixos/tests/mpd.nix [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
   let
     track = pkgs.fetchurl {
       # Sourced from http://freemusicarchive.org/music/Blue_Wave_Theory/Surf_Music_Month_Challenge/Skyhawk_Beach_fade_in
-      # License: http://creativecommons.org/licenses/by-sa/4.0/
 
       name = "Blue_Wave_Theory-Skyhawk_Beach.mp3";
       url = "https://freemusicarchive.org/file/music/ccCommunity/Blue_Wave_Theory/Surf_Music_Month_Challenge/Blue_Wave_Theory_-_04_-_Skyhawk_Beach.mp3";
-      sha256 = "0xw417bxkx4gqqy139bb21yldi37xx8xjfxrwaqa0gyw19dl6mgp";
+      hash = "sha256-91VDWwrcP6Cw4rk72VHvZ8RGfRBrpRE8xo/02dcJhHc=";
+      meta.license = lib.licenses.cc-by-nc-sa-40;
```
Comment (Scrumplex) [inline nixos/tests/mpd.nix:10]: http://freemusicarchive.org/music/Blue_Wave_Theory/Surf_Music_Month_Challenge/Skyhawk_Beach_fade_in specifies that the license is CC BY SA 4.0. Setting this to CC BY NC SA 4.0 makes the test closure unfree

```suggestion
      meta.license = lib.licenses.cc-by-sa-40;
```

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347387`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>clerk</li>
    <li>mpc</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**